### PR TITLE
fixed python bindings signed/unsigned regs/mem

### PR DESCRIPTION
**Pull request by mrphrazer**
seems to fix the [unsigned issues](https://github.com/unicorn-engine/unicorn/issues/654) however, I am not sure if I "fixed" something that was correct, please review before making things worse

### CODE REVIEW

File: bindings/python/unicorn/unicorn.py [hunk 1]
```diff
@@ -89,8 +89,8 @@ def _setup_prototype(lib, fname, restype, *argtypes):
 _setup_prototype(_uc, "uc_close", ucerr, uc_engine)
 _setup_prototype(_uc, "uc_strerror", ctypes.c_char_p, ucerr)
 _setup_prototype(_uc, "uc_errno", ucerr, uc_engine)
-_setup_prototype(_uc, "uc_reg_read", ucerr, uc_engine, ctypes.c_int, ctypes.c_void_p)
-_setup_prototype(_uc, "uc_reg_write", ucerr, uc_engine, ctypes.c_int, ctypes.c_void_p)
+_setup_prototype(_uc, "uc_reg_read", ucerr, uc_engine, ctypes.c_uint64, ctypes.c_void_p)
```
Comment (aquynh) [inline bindings/python/unicorn/unicorn.py]: the original prototype of uc_reg_read() is like this:

    uc_err uc_reg_read(uc_engine *uc, int regid, void *value);

so you cannot use c_uint64 here, but just c_uint.

Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: this is incorrect. regid is just an index.

File: bindings/python/unicorn/unicorn.py [hunk 2]
```diff
@@ -89,8 +89,8 @@ def _setup_prototype(lib, fname, restype, *argtypes):
...
+_setup_prototype(_uc, "uc_reg_write", ucerr, uc_engine, ctypes.c_uint64, ctypes.c_void_p)
```
Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: again, this is just an index. please check the prototypes. they live in `include/unicorn/unicorn.h`

File: bindings/python/unicorn/unicorn.py [hunk 3]
```diff
@@ -114,11 +114,11 @@ def _setup_prototype(lib, fname, restype, *argtypes):
 UC_HOOK_CODE_CB = ctypes.CFUNCTYPE(None, uc_engine, ctypes.c_uint64, ctypes.c_size_t, ctypes.c_void_p)
 UC_HOOK_MEM_INVALID_CB = ctypes.CFUNCTYPE(
     ctypes.c_bool, uc_engine, ctypes.c_int,
```
Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: read the prototypes here 

```c
typedef void (*uc_cb_hookmem_t)(uc_engine *uc, uc_mem_type type,                
        uint64_t address, int size, int64_t value, void *user_data);
```

**Comment by mrphrazer** [comment]
thanks, fixed!

File: bindings/python/unicorn/unicorn.py [hunk 4]
```diff
@@ -114,11 +114,11 @@ def _setup_prototype(lib, fname, restype, *argtypes):
...
-    ctypes.c_uint64, ctypes.c_int, ctypes.c_int64, ctypes.c_void_p
+    ctypes.c_uint64, ctypes.c_uint, ctypes.c_uint64, ctypes.c_void_p
 )
 UC_HOOK_MEM_ACCESS_CB = ctypes.CFUNCTYPE(
     None, uc_engine, ctypes.c_int,
-    ctypes.c_uint64, ctypes.c_int, ctypes.c_int64, ctypes.c_void_p
+    ctypes.c_uint64, ctypes.c_uint, ctypes.c_uint64, ctypes.c_void_p
```
Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: same, consult the prototypes.

File: bindings/python/unicorn/unicorn.py [hunk 5]
```diff
@@ -255,7 +255,7 @@ def reg_read(self, reg_id):
                 return reg.low_qword | (reg.high_qword << 64)
 
         # read to 64bit number to be safe
```
Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: this comment is misleading... can we do something about that?

File: bindings/python/unicorn/unicorn.py [hunk 6]
```diff
@@ -284,7 +284,7 @@ def reg_write(self, reg_id, value):
 
         if reg is None:
             # convert to 64bit number to be safe
```
Comment (lowlyw) [inline bindings/python/unicorn/unicorn.py]: same. Nguyen?

File: bindings/python/unicorn/unicorn.py (hunk 3 above)
Comment (aquynh) [inline bindings/python/unicorn/unicorn.py]: what do you mean, @lowlyw? it is hard to follow this. if you find something wrong, can you open a new issue?

Comment (aquynh) [inline bindings/python/unicorn/unicorn.py]: you are right, @mrphrazer should not change the API prototype.

**Comment by aquynh** [comment]
merged, but i reverted the changes on the API prototypes, thanks!
